### PR TITLE
[SPARK-43192] [CONNECT] Remove user agent charset validation

### PR DESCRIPTION
**Pull request by nija-at**
### Why are the changes needed?

The current validation on charset is restrictive. It does not allow
blank space and digits. It's common for user agent strings to contain
these characters.

Secondly, it restricts the length to stay under 200 characters.
The limit to 200 characters was mostly something that was done as a
simple protection mechanism. We've looked into different specifications
for what could be part of the user agent and longer user agents are
common. Increase this restriction to 2KB.
The server should enforce restrictions on its side, but we would still like
to keep the restriction as fallback protection, but allow large values.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Unit tests

### CODE REVIEW

File: python/pyspark/sql/connect/client.py [hunk 1]
```diff
@@ -269,18 +269,9 @@ def userAgent(self) -> str:
         user_agent : str
             The user_agent parameter specified in the connection string,
             or "_SPARK_CONNECT_PYTHON" when not specified.
+            The returned value will be percent encoded.
         """
-        user_agent = self.params.get(ChannelBuilder.PARAM_USER_AGENT, "_SPARK_CONNECT_PYTHON")
-        allowed_chars = string.ascii_letters + string.punctuation
-        if len(user_agent) > 200:
```
Comment (grundprinzip) [inline python/pyspark/sql/connect/client.py:275]: should we not assert on some length? maybe 1204?

**Review by grundprinzip — APPROVED** [review]

**Comment by amaliujia** [comment]
Just curious 

```
Remove this restriction from the client and let the service enforce
length limits.
```
Is the server already enforcing the length limits or you plan to have a follow-up to add that.

**Comment by grundprinzip** [comment]
There is no server-side validation. The limit to 200 characters was mostly something that was done as a simple protection mechanism. We've looked into different specifications for what could be part of the user agent and the conclusion was to not have a limit at this point.

**Review by amaliujia — APPROVED** [review]

File: python/pyspark/sql/connect/client.py [hunk 2]
```diff
@@ -269,18 +267,15 @@ def userAgent(self) -> str:
...
-            )
-        if set(user_agent).difference(allowed_chars):
+        ua_len = len(urllib.parse.quote(user_agent))
+        if ua_len > 2048:
             raise SparkConnectException(
-                "Only alphanumeric and common punctuations are allowed for 'user_agent'"
+                f"'user_agent' parameter should not exceed 2048 characters, found {len} characters."
             )
-        return user_agent
+        return self.params.get(ChannelBuilder.PARAM_USER_AGENT, "_SPARK_CONNECT_PYTHON")
```
Comment (HyukjinKwon) [inline python/pyspark/sql/connect/client.py:278]: Why don't we just return `user_agent`?

```suggestion
        return user_agent
```

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

(since the last change is basically no-op)

**Pull request closed without merging**